deepset_cloud_sdk/_api/config.py

```diff
@@ -12,48 +12,81 @@
 
 def load_environment() -> bool:
-    """Load environment variables from the .env file.
+    """Load environment variables using a cascading fallback model.
 
-    If an .env file exists in the current directory, load the environment variables from there.
-    Otherwise, load the environment variables from the .env file in the home directory. You can create this file using the CLI
-    by running `deepset-cloud login` in the terminal.
+    1. Load local .env file in current directory if it exists
+    2. For additional variables, load from global ~/.deepset-cloud/.env
+    3. Environment variables can override both
 
-    :return: True if the environment variables were loaded successfully, False otherwise.
+    :return: True if required environment variables were loaded successfully, False otherwise.
     """
-    successfully_loded_env: bool = False
+    successfully_loaded_env: bool = False
     current_path_env = os.path.join(os.getcwd(), ".env")
-    if os.path.isfile(current_path_env):
-        # Load the environment variables from the .env file in the current directory
-        successfully_loded_env = load_dotenv(current_path_env)
-        return successfully_loded_env
+    global_env_exists = os.path.isfile(ENV_FILE_PATH)
+    local_env_exists = os.path.isfile(current_path_env)
+
+    # First load local config
+    if local_env_exists:
+        successfully_loaded_env = load_dotenv(current_path_env)
+        logger.debug(f"Loaded local .env file at {current_path_env}.")
+
+    # Then load global config only for variables which do not exist in the local .env file
+    if global_env_exists:
+        global_loaded = load_dotenv(ENV_FILE_PATH, override=False)
+        if global_loaded:
+            successfully_loaded_env = True
+            logger.debug(
+                f"Loaded global .env file at {ENV_FILE_PATH} for variables which do not exist "
+                "in the local .env file."
+            )
+
+    if not successfully_loaded_env:
+        logger.warning(
+            "No .env files found. Run `deepset-cloud login` to create a global configuration file. "
+            "You can also create a custom local .env file in your project directory."
+        )
+        return successfully_loaded_env
+
+    # Check for required environment variables
+    required_vars = ["API_KEY", "API_URL", "DEFAULT_WORKSPACE_NAME"]
+    missing_vars = [var for var in required_vars if not os.getenv(var)]
+
+    if missing_vars:
+        successfully_loaded_env = False
+        logger.warning(
+            f"Missing required environment variables: {', '.join(missing_vars)}. "
+            "Please run `deepset-cloud login` to set up your configuration or set these variables manually."
+        )
     else:
-        # Load the environment variables from the .env file in the home directory
-        successfully_loded_env = load_dotenv(ENV_FILE_PATH)
-        return successfully_loded_env
+        logger.info("Environment variables loaded successfully.")
```

Comment:
I think it'd change the `debug`/`info` stuff around. Reasoning:

* the fact that the environment variables where loaded is not interesting
* It is interesting _where_ the environment variables where loaded from though.